fix Issue 21962 - importC: Empty enums are accepted as valid code

ibuclaw opened the pull request:
@WalterBright

ibuclaw committed:
fix Issue 21962 - importC: Empty enums are accepted as valid code

thewilsonator commented on test/fail_compilation/failcstuff1.d:
```diff
@@ -0,0 +1,8 @@
+---
+fail_compilation/imports/cstuff1.c(5): Error: no members for `enum empty_enum`
+fail_compilation/imports/cstuff1.c(6): Error: no members for anonymous enum
+---
```
Windows boxen fail with:
```
expected:
----
fail_compilation/imports/cstuff1.c(5): Error: no members for `enum empty_enum`
fail_compilation/imports/cstuff1.c(6): Error: no members for anonymous enum
----
actual:
----
fail_compilation\imports\cstuff1.c(5): Error: no members for `enum empty_enum`
fail_compilation\imports\cstuff1.c(6): Error: no members for anonymous enum
```

urgh.

UplinkCoder replied:
The test script probably does not apply the path-rewrite-regex on .c files.

ibuclaw committed:
d_do_test: Unify directory paths on C source files

The introduction of importC means C sources can now be compiled with a D compiler.
